e2e/redis_test.go

```diff
@@ -0,0 +1,342 @@
+	"context"
+	"io"
+	"strings"
+	"testing"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	controllerruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/e2e-framework/klient"
+	"sigs.k8s.io/e2e-framework/klient/k8s"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+	"sigs.k8s.io/e2e-framework/support/utils"
+
+	spinapps_v1alpha1 "github.com/spinkube/spin-operator/api/v1alpha1"
+	"github.com/spinkube/spin-operator/e2e/helper"
+	"github.com/spinkube/spin-operator/internal/generics"
+)
+
+// TestShimRedis checks that the shim has basic runtime config support
+// by being able to integrate with redis as the backend for a kv store
+func TestShimRedis(t *testing.T) {
+	var client klient.Client
+
+	// TODO: Use an image from a sample app in this repository
+	appImage := "ghcr.io/calebschoepp/spin-checklist:v0.1.0"
+	testSpinAppName := "test-shim-redis"
+
+	defaultTest := features.New("default and most minimal setup").
+		Setup(func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+
+			client = cfg.Client()
+
+			if err := spinapps_v1alpha1.AddToScheme(client.Resources(testNamespace).GetScheme()); err != nil {
+				t.Fatalf("failed to register the spinapps_v1alpha1 types with Kubernetes scheme: %s", err)
+			}
+
+			return ctx
+		}).
+		Assess("redis is setup", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			if err := client.Resources().Create(ctx, redisDeployment()); err != nil {
+				t.Fatalf("Failed to create redis deployment: %s", err)
+			}
+
+			if err := client.Resources().Create(ctx, redisService()); err != nil {
+				t.Fatalf("Failed to create redis service: %s", err)
+			}
+
+			return ctx
+		}).
+		Assess("spin app custom resource is created", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			testSpinApp := newSpinAppUsingRedis(testSpinAppName, appImage, "containerd-shim-spin")
+
+			if err := client.Resources().Create(ctx, testSpinApp); err != nil {
+				t.Fatalf("Failed to create spinapp: %s", err)
+			}
+			// wait for spinapp to be created
+			if err := wait.For(
+				conditions.New(client.Resources()).ResourceMatch(testSpinApp, func(object k8s.Object) bool {
+					return true
+				}),
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(30*time.Second),
+			); err != nil {
+				t.Fatal(err)
+			}
+			return ctx
+		}).
+		Assess("spin app deployment and service are available", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// wait for deployment to be ready
+			if err := wait.For(
+				conditions.New(client.Resources()).DeploymentAvailable(testSpinAppName, testNamespace),
+				wait.WithTimeout(3*time.Minute),
+				wait.WithInterval(30*time.Second),
```

Comment:
```suggestion
				wait.WithInterval(time.Second),
```

30 seconds between checks is a _really_ long time (especially as the first check doesn't happen until after that interval) - and the check is ~free, so cut this down.